File: src/org/translateai/crypto/utils/io/FileUtils.java
package org.translateai.crypto.utils.io;

import java.io.IOException;
import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Random;

public class FileUtils {
    private FileUtils() {
        //Prevent instantiation. This class contains only static methods.
    }
    public static byte[] readFile(String folderPath, String fileName) {
        Path path = Paths.get(folderPath, fileName);

        try {
            return Files.readAllBytes(path);
        } catch (IOException e) {
            throw new FileProcessingException("Error reading file: " + path, e);
        }
    }

    public static void writeFile(String folderPath, String fileName, byte[] data) {
        Path path = Paths.get(folderPath, fileName);

        try {
            Files.write(path, data);
        } catch (IOException e) {
            throw new FileProcessingException("Error writing file: " + path, e);
        }
    }

    public static String createTestSubFolder(String parentFolder) {
        String name = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss-")
                .format(new Date()) + new Random().nextInt(10000);
        boolean mkDirsResult = new File(parentFolder, name).mkdirs();
        String fullPath = Paths.get(parentFolder, name).toString();

        if (!mkDirsResult) {
            throw new FileProcessingException("Directory already exists: " + fullPath, null);
        }

        return fullPath;
    }

    public static boolean areEqual(String folder1, String file1,
                                String folder2, String file2) {
        Path path1 = Paths.get(folder1, file1);
        Path path2 = Paths.get(folder2, file2);

        try {
            return Files.mismatch(path1, path2) == -1;
        } catch (IOException e) {
            throw new FileProcessingException("Error comparing files: " +
                    path1 + " and " +
                    path2, e);
        }
    }
}

File: translateai-generated-code-updated/src/org/translateai/crypto/utils/io/FileUtils.py
#5 imports below were added manually
from org.translateai.crypto.utils.io.FileProcessingException import FileProcessingException
from pathlib import Path
import datetime
import random
import os

class FileUtils:
    def __init__(self):
        raise TypeError("FileUtils cannot be instantiated")

    @staticmethod
    def readFile(folderPath, fileName):
        path = Path(f"{folderPath}/{fileName}")
        try:
            with open(path, 'rb') as f:
                return f.read()
        except Exception as e:
            raise FileProcessingException(f"Error reading file: {path}", e)

    @staticmethod
    def writeFile(folderPath, fileName, data):
        path = Path(f"{folderPath}/{fileName}")
        try:
            with open(path, 'wb') as f:
                f.write(data)
        except Exception as e:
            raise FileProcessingException(f"Error writing file: {path}", e)

    @staticmethod
    def createTestSubFolder(parentFolder):
        #added missing dash after %S - minor fix for the line below
        name = f"{datetime.datetime.now().strftime('%Y-%m-%d-%H-%M-%S-')}{random.randint(1000, 9999)}"
        new_folder = Path(f"{parentFolder}/{name}")
        try:
            os.makedirs(new_folder)
        except FileExistsError:
            raise FileProcessingException(f"Directory already exists: {new_folder}", None)
        return str(new_folder)

    @staticmethod
    def areEqual(folder1, file1, folder2, file2):
        path1 = Path(f"{folder1}/{file1}")
        path2 = Path(f"{folder2}/{file2}")
        try:
            #Not exactly the way to compare files that was used originally, but OK
            return hash(open(path1, 'rb').read()) == hash(open(path2, 'rb').read())
        except Exception as e:
            raise FileProcessingException(f"Error comparing files: {path1} and {path2}", e)
